python/pydantic_core/_pydantic_core.pyi

```diff
@@ -305,6 +305,7 @@ class SchemaSerializer:
         exclude_unset: bool = False,
         exclude_defaults: bool = False,
         exclude_none: bool = False,
+        sort_keys: Literal['recursive', 'top-level', 'unsorted'] = 'unsorted',
```

Comment:
do we really need these three options? surely if we add this feature it should match python's `json.dumps` behaviour and be `sort_keys: bool = False`.

Reply:
There are scenarios where you'd want to only sort the top level, e.g. the trick we were doing with our JSON data in FusionFire.

But I also agree that if no one is asking for it we shouldn't implement extra stuff.

How does it sound if for now we do `sort_keys: bool = False` and then if needed in the future we can do `sort_keys: bool | Literal['top-level'] = False`.
We reduce the complexity and make a nicer signature for now but can still add the top-level option in the future, even if it makes the signature a bit weird.

@samuelcolvin @aezomz wdyt?

Reply:
agreed, that's what we've done for previous flags.

Reply:
Sure, let me check the feature for sort_keys on json.dumps.
I only want to do `sort_keys` on `model_dump_json` thats why am here 😆 
Let me try to implement this.

Reply:
Thank you so much @aezomz, you are being incredibly patient and helpful 🙏🏻

Reply:
Hi Adrian, please review when ur free. I think David fixed the wasm build test already.